### PR TITLE
Stop scrolling to top of article when window is resized

### PR DESCRIPTION
**Pull request by dennisschagt**
Fixes https://github.com/newsboat/newsboat/issues/2073
Fixes https://github.com/newsboat/newsboat/issues/1298

### CODE REVIEW

**Comment by dennisschagt** [comment]
From https://github.com/newsboat/newsboat/issues/1298#issuecomment-731631997:
> Just resetting the position is an easy way out. Coming up with a more intelligent algorithm sounds like a pretty interesting problem, actually :)

There might be some intelligent algorithm possible, but implementing it doesn't sound interesting to me :)
For now, I just changed back to the old behavior of keeping the same scroll offset (same line number on top) when resizing the terminal.

File: src/textviewwidget.cpp [hunk 1]
```diff
@@ -26,6 +26,17 @@ void TextviewWidget::stfl_replace_lines(std::uint32_t number_of_lines,
...
+	if (num_lines == 0) {
+		set_scroll_offset(0);
+	} else {
+		const std::uint32_t max_offset = num_lines - 1;
+		const std::uint32_t current_offset = get_scroll_offset();
+
+		if (current_offset > max_offset) {
+			set_scroll_offset(max_offset);
+		}
+	}
```
Comment (Minoru) [inline src/textviewwidget.cpp:39]: Could you please add a test for this?

**Review by Minoru — CHANGES_REQUESTED** [review]
Glad you're back Dennis :) The changes look good, but please add a test for new `TextView` behaviour; I worked on `ListWidget` recently, and it reminded me that it's better to get tests while the reasoning behind the code is still fresh in the mind.

File: src/textviewwidget.cpp (hunk 1 above)
Comment (dennisschagt) [reply src/textviewwidget.cpp:39]: Sure, done

File: test/textviewwidget.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	}
+
+	SECTION("scroll_to_bottom() moves scroll location to last line") {
+		widget.scroll_to_bottom();
+		// Scroll offset is set in such a way that one line past the end is visible at the bottom. Required scroll offset depends on the height of the textview (hardcoded: 5 lines)
+		// Given 10 items in the list, scroll offset 6 shows line seven as first line, followed by eight, nine, ten, and an empty line.
+		REQUIRE(widget.get_scroll_offset() == 6);
+	}
+
+	SECTION("scroll_page_up() moves up just enough such that there is a single line of overlap between old and niw view") {
```
Comment (Minoru) [inline test/textviewwidget.cpp:132]:
```suggestion
	SECTION("scroll_page_up() moves up just enough such that there is a single line of overlap between old and new view") {
```

File: test/textviewwidget.cpp [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		REQUIRE(widget.get_scroll_offset() == 6);
+
+		widget.scroll_page_up();
+		REQUIRE(widget.get_scroll_offset() == 2);
+
+		widget.scroll_page_up();
+		REQUIRE(widget.get_scroll_offset() == 0);
+	}
+
+	SECTION("scroll_page_down() moves down just enough such that there is a single line of overlap between old and niw view") {
```
Comment (Minoru) [inline test/textviewwidget.cpp:143]:
```suggestion
	SECTION("scroll_page_down() moves down just enough such that there is a single line of overlap between old and new view") {
```

**Review by Minoru — CHANGES_REQUESTED** [review]

File: test/textviewwidget.cpp [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+
+		REQUIRE(widget.get_scroll_offset() == 2);
+	}
+}
+
+TEST_CASE("basic movements work as expected", "[TextviewWidget]")
+{
+	ConfigPaths paths;
+	Controller c(paths);
+	newsboat::View v(&c);
```
[the remark this answers is not in the document]
Comment (dennisschagt) [reply test/textviewwidget.cpp]: I just realized it is not even necessary at all.
The only place which activates the terminal/ncurses "application mode" is the `form.run(-3)` call.
So, just adding a single `Stfl::reset()` call directly after that works fine.

**Review by Minoru — APPROVED** [review]

**Comment by Minoru** [comment]
Thank you!

**Pull request merged**